### DISCUSSION TITLE
Keda Implementation ways

### DISCUSSION THREAD

**Discussion by pavan-chakka** [kedacore/keda#2994] [Q&A / Need Help] [answered]
Hello,

In our org, we use Azure as our cloud services, and we have several teams who work on various applications/Azure functions. In Azure, we have non prod AKS cluster and Prod AKS cluster. We run our applications dev, test, QA APPs on non prod AKS cluster and prod on Prod AKS cluster. Each AKS cluster have namespace allocated for each team and they run their apps in PODS in their allocated namespaces. A team member will have access only to their name space and not to other team namespace.

Several of our teams have Batch jobs and functions which need to be triggered with an Event occurrence. So, we want to use KEDA to scale based on volume of events. We want to see if it is doable to have KEDA in each name space of an AKS cluster and deploy their functions/batch jobs in their name space and scale them based on events.

Our understanding is that we can have only one KEDA installed per AKS cluster. If this is true, how many functions KEDA can handle per AKS cluster, and how can we organize Azure functions/APPS/Batch processes per team/ per project.

If we can have multiple KEDAs per AKS (preferably one KEDA per name space), then how to implement this.

Please advise on best architecting ways around KEDA, and how other organizations are implementing it. 

Thanks.

**Comment by tomkerkhove** [comment, marked as answer]
Unfortunately, we don't allow you to support KEDA in multiple clusters (see https://github.com/kedacore/keda/issues/2654)

This should, however, not be a blocker for your scenario as you can scale as many workloads as you want (we have end-users with 1000's of ScaledObjects).

That is also why we provide `TriggerAuthentication` which is scoped to a namespace and `ClusterTriggerAuthentication` that is cross-cluster.

**Reply by pavan-chakka** [reply]
Thank you Tom for taking time to respond. I got it, and thanks for clarifying.

**Reply by tomkerkhove** [reply]
Happy to help!
